### PR TITLE
impl(pubsub): add ack id and subscription to lease interface

### PR DESCRIPTION
**Pull request by alevenberg**
Part of https://github.com/googleapis/google-cloud-cpp/issues/13298

This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13367)

### CODE REVIEW

File: google/cloud/pubsub/internal/pull_lease_manager.h [hunk 1]
```diff
@@ -45,6 +45,12 @@ class PullLeaseManager {
   virtual future<Status> ExtendLease(std::shared_ptr<SubscriberStub> stub,
                                      std::chrono::system_clock::time_point now,
                                      std::chrono::seconds extension) = 0;
+
+  virtual std::string ack_id() const { return ""; }
+
+  virtual pubsub::Subscription subscription() const {
+    return pubsub::Subscription{};
+  }
```
Comment (coryan) [inline google/cloud/pubsub/internal/pull_lease_manager.h:53]: Having default implementations is good.  This is hiding the need to add the corresponding `MOCK_METHOD()` in the mock class, right?

**Review by coryan — CHANGES_REQUESTED** [review]

File: google/cloud/pubsub/internal/pull_lease_manager.h (hunk 1 above)
Comment (alevenberg) [reply google/cloud/pubsub/internal/pull_lease_manager.h:53]: No, I think we still need to mock it for our tests. The subscription is part of the span name.

Comment (alevenberg) [reply google/cloud/pubsub/internal/pull_lease_manager.h:53]: If we left it unmocked, all the tests would have something like --invalid-- extend, which i guess isn't the worst. The mock is also in testing, not mocks.

Comment (coryan) [reply google/cloud/pubsub/internal/pull_lease_manager.h:53]: I am not sure if you are saying that we should leave it as-is or not.  I would just point out that if you ever wanted to add a test that needed to mock these functions you may be surprised by the errors.

Comment (alevenberg) [reply google/cloud/pubsub/internal/pull_lease_manager.h:53]: I was saying we should leave it. I mock'd them in https://github.com/googleapis/google-cloud-cpp/pull/13369/ and it seems to be working. I had some other errors I needed to fix, but it runs.

Comment (coryan) [reply google/cloud/pubsub/internal/pull_lease_manager.h:53]: That PR adds the `MOCK_METHOD()` calls.  It seems more organized to add them here, even if they go unused at first.  And I guess I missed the same comment for `ExtendLease`.  Your call.

**Review by coryan — APPROVED** [review]

File: google/cloud/pubsub/internal/pull_lease_manager.h (hunk 1 above)
Comment (alevenberg) [reply google/cloud/pubsub/internal/pull_lease_manager.h:53]: Gonna add them in the next one.

**Pull request merged**